html/app.js

```diff
@@ -38,32 +38,31 @@ export class MicEnumerator {
   async force_permission_prompt() {
     // In order to enumerate devices, we must first force a permission prompt by opening a device and then closing it again.
     // See: https://stackoverflow.com/questions/60297972/navigator-mediadevices-enumeratedevices-returns-empty-labels
-    var stream = await navigator.mediaDevices.getUserMedia({ audio: true });
-    close_stream(stream);
+    try {
+      var stream = await navigator.mediaDevices.getUserMedia({ audio: true });
+      close_stream(stream);
+      return;
+    } catch (e) {
+      // Do nothing. Our only goal here is to prompt the user to allow their mic if they have one. Any failure is irrelevant to us.
+      return;
+    }
   }
 
   async wait_for_mic_permissions() {
-    var perm_status = await navigator.permissions.query({name: "microphone"}).catch(() => null);
-    if (!perm_status) {
-      this.force_permission_prompt();
-      return;
-    }
+    // First, check if we already have permission, without needing to do anything.
+    var perm_status = await navigator.permissions.query({name: "microphone"}).catch(() => {});
     if (perm_status.state == "granted" || perm_status.state == "denied") {
+      // Denied means we will fail, but also there's nothing more we can do about it.
       return;
     }
-    this.force_permission_prompt();
-    return new Promise((resolve, reject) => {
-      perm_status.onchange = (e) => {
-        if (e.target.state == "granted" || e.target.state == "denied") {
-          resolve();
-        }
-      }
-    });
+    // This can't fail, but eat errors just to be sure, because we don't care.
+    return this.force_permission_prompt().catch(() => undefined);
```

Comment:
I am not in favor of the "eat errors just to be sure" thing. We are already eating the errors that can be thrown by the platform-native APIs; any errors that nonetheless make it out here are probably the result of bugs in our code (which could be as simple as typos), and suppressing those is likely to just lead to another crash later, with that latter crash being hard to debug because we lost the stack trace of the first failure.